src/tests/universal-nrp-e2e/README.md

```diff
@@ -0,0 +1,281 @@
+# Universal NRP End-to-End Tests Local/GitHub
+This directory contains tests for the universal NRP both for local/onprem and GitHub workflows. There is a GitHub workflow which performs these tests but only for modern distros (excluding EOL distributions eg. centos 7-8, ubuntu 18.04-20.04, etc.). We do support EOL testing on our infrastructure, as such, we've developed some tools in order for us to test locally.
+
+The following bash tools have been created to enable testing across any linux-based OS:
+ - `StartLocalTest.sh`: Peforms tests locally with a given Azure Policy package. This is meant to be used directly on the machine being tested (See [Testing on VMs](#testing-on-vms)).
+ - `StartVMTest.sh`: Uses QEMU with a given cloud based image (cloud-init enabled) and performs the tests on the VM (See [Testing directly on target machine](#testing-directly-on-target-machine)).
+ - `StartTests.sh` (Microsoft Internal Only): Uses the _StartVMTest.sh_ above to orchestrate tests on images not being tested by GitrHub workflows. These are internal as they leverage Azure Blob Storage for storing the disk images (See [Testing all supported distributions (Internal-Only)](#testing-all-supported-distributions-internal-only).
+
+# Testing on VMs
+To perform tests on VMs the `StartVMTest.sh` allows you to perform tests on specific Azure Policy packages targetting a particular VM disk image. QEMU is used as the virtualization platform and allows us to target linux distributions shared as raw/qcow2 images (See [VM Image Sources](#vm-image-sources) for compatible images). Images are booted using cloud-init in order to provision the user and help orchestrate tests, collect logs and test reports back to the QEMU host.
+
+## Example
+The following example performs tests on an [Ubuntu Noble cloud image](https://cloud-images.ubuntu.com/noble/current/) on the AzureLinuxBaseline.zip (built in directory tree) which has 168 resources defined.
+```sh
```

Comment:
Is the intent to always run this on a Linux host? Will this work on a bash client for Windows (maybe WSL etc.)?

Reply:
Yes its meant to be run on a linux host. This works quite well in WSL, ive been using it this way and even spawn 10 concurrent VMs with 2Gb each without any issues.

Reply:
Awesome! Great that this works with WSL!